src/Tokenizer/Transformer/SquareBraceTransformer.php

```diff
@@ -44,6 +44,15 @@ public function getCustomTokens()
         ];
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // must run after CurlyBraceTransformer
```

Comment:
both `BraceClassInstantiationTransformer` and `CurlyBraceTransformer` check for `CT::T_ARRAY_SQUARE_BRACE_CLOSE` which comes from this transformer, so there is a priority issue...